### PR TITLE
Feat/add renewal page

### LINKED ISSUE #394: Domain Renewal page
Figma link : https://www.figma.com/file/S1UKYgWewNqNHZFAaBUilG/Starknet-ID-design?type=design&node-id=1565%3A660&mode=design&t=GGc5OHbzXM1MzXMK-1

image

Here is an example of an email page that a user receives when their domain is about to expire. A link represented by a button allows them to quickly renew their domain with a 20% discount.

image

By clicking on the 'Renew my domain' button, the user is redirected to the renewal page with various input fields. The discount is displayed next to the total due.

### LINKED ISSUE #438: Bug: clicking on id card doesn't open it, you need to click on the avatar

### PR DESCRIPTION
**Pull request by fricoben**
close #394 

this also close #438 

This page permits to addition of a renewal page that will permit anybody to renew their domains with a big multicall. Moreover, I changed some organization of the front-end with `write_confirmation` transactions to be more clean.

Here are the things to test before merging (done by @fricoben) : 
- [x] Domain renew on modal 
- [x] Multicall domain renewal (one year by default here)
- [x] VAT transfer with multi-call domain renewal
- [ ] WriteVerifierData with twitter, discord, GitHub and POP

This PR can be an example of managing callData in the future. Ideally, we will move all the callData to the new folder created in this PR. 

We'll need 2 reviews here and it's very important to be cautious cause an error would create a lot of problems in production.

### CODE REVIEW

File: components/domains/renewal.tsx [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+              value={email}
+              onChange={(e) => changeEmail(e.target.value)}
+              color="secondary"
+              error={emailError}
+              errorMessage={"Please enter a valid email address"}
+              type="email"
+            />
+            <UsForm
+              isUsResident={isUsResident}
+              onUsResidentChange={() => setIsUsResident(!isUsResident)}
```
Comment (Th0rgal) [inline components/domains/renewal.tsx:210]: it's up to you to decide but when I update a state value using its existing value I prefer to use the function form, so setIsUsResident(prev => !prev) for example.

File: components/domains/renewal.tsx [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+            >
+              {!termsBox
+                ? "Please accept terms & policies"
+                : isUsResident && !usState
+                ? "We need your US State"
+                : invalidBalance
+                ? "You don't have enough eth"
+                : emailError
+                ? "Enter a valid Email"
+                : "Register my domain"}
```
Comment (irisdv) [inline components/domains/renewal.tsx]: Replace by `Renew my domain` or even `domains`

**Review by Th0rgal — CHANGES_REQUESTED** [review]
Issues:
- It is loading forever when I am not logged in.
- The page is `/[domain_variable].tsx` but doesn't use domain_variable

Suggestions:
- It could be nice to have a way to add new domains to the list, even if you don't own them (so that if I want to renew a domain for argent, partners or I one I deposited on a contract, I can do it)
- In the future having the option to toggle auto_renew would be great

Questions:
- How should users access this page? Is there a link in the website that I missed?

File: components/domains/renewalDomainsBox.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  }, [address]);
+
+  const handleCheckboxChange = (domain: string) => {
+    setSelectedDomains((prevState) => ({
+      ...prevState,
+      [domain]: !prevState?.[domain],
+    }));
+  };
+
+  return isLoading ? (
```
Comment (irisdv) [inline components/domains/renewalDomainsBox.tsx:60]: When the user is not connected, the domain box is always in loading mode, it's not really clear for the user that he's not connected

**Review by irisdv — CHANGES_REQUESTED** [review]
I left a few comment in the code. Also if no domains are selected I think the button should be grey, otherwise it can send a tx with a transfer and approve of 0 :
Capture d’écran 2023-10-10 à 18 32 09

I don't understand what `domainToRenew` in the renewal page is used for. I tried with different domains and subdomains but it didn't have any effect.

I think the button in the identity page should also link to this page, because it's nicer but also because you can renew all your domains at once which is more practical than going through each identity. 

Why not show the expiration date as well in the renewal page ?

**Comment by fricoben** [comment]
this also close #438

**Review by irisdv — APPROVED** [review]
lgtm!

File: components/UI/backButton.tsx [hunk 4]
```diff
@@ -7,11 +7,29 @@ type BackButtonProps = {
...
+
+  return (
+    <button
+      onClick={onClick}
+      onMouseEnter={() => setOnHover(true)}
+      onMouseLeave={() => setOnHover(false)}
+      className={styles.backButton}
+    >
+      <ArrowLeftIcon
+        color={onHover ? "#402d28bb" : theme.palette.secondary.main}
```
Comment (Th0rgal) [inline components/UI/backButton.tsx]: Using javascript for hover effects is a bad practice. I would suggest something like:
```css
.backButton {
  &:hover .arrowIcon {
    color: #402d28bb;
  }
  
  &:hover .backButtonLabel {
    /* Your hover styles for label */
  }
}
```

**Review by Th0rgal — APPROVED** [review]

**Pull request merged**